File: PRO/C++/UT02/Practica1-AlejandroHernandezEscudero/factorial.cpp
#include <stdio.h>
#include <conio.h>
#ifdef _WIN32
#include <windows.h>
#endif
int main()
{
    int numero;
    int fact = 1;
    do
    {
        printf("Escribe un numero mayor que 1 para realizar el factorial: ");
        scanf("%d", &numero);
  
        
    } while (numero < 1);

    for (int i = 1; i <= numero; i++)
    {
        fact = fact * i;
    }

    printf("El factorial de %d es %d", numero , fact);
    
    return (fact);
    system("cls");
}

File: ETS/ejercicios compiladores/factorial.java
import java.util.Scanner;

class factorial {
	public static void main(String[] args) {
		int numero;
		long fact = 1;

		do {
			System.out.println("Escribe un numero mayor que 0");
			Scanner teclado = new Scanner(System.in);
			numero = teclado.nextInt();

		} while (numero <= 0);

		for (int i = 1; i <= numero; i++) {
			fact = fact * i;
		}
		System.out.println(("El factorial de " + numero + " es: ") + fact);
	}

}